actionlist/app/reducers/actionItems.js

```diff
@@ -108,10 +119,44 @@ const filterActionList = (state, action) => {
 const resetFilters = (state) => {
   return {
     ...state,
+    drawerExpanded: false,
     filterStatus,
   };
 };
 
+const dateFilterChange = (state, action) => {
+  switch (action.payload.title) {
```

Comment:
What if, instead of an unwieldy switch with lots of duplicate code, you just returned
```es6
{
  ...state,
  drawerExpanded: false,
  filterStatus: {
    ...state.filterStatus,
    [action.payload.title]: {
      ...state.filterStatus[action.payload.title],
      [action.payload.position]: action.payload.date,
    },
  },
}
```